src/points/PointsHistoryBottomSheet.tsx

```diff
@@ -0,0 +1,182 @@
+    ValoraAnalytics.track(PointsEvents.points_screen_activity_try_again_press)
+    dispatch(
+      getHistoryStarted({
+        getNextPage: false,
+      })
+    )
+  }
+
+  const fetchMoreHistory = () => {
+    ValoraAnalytics.track(PointsEvents.points_screen_activity_fetch_more)
+    dispatch(
+      getHistoryStarted({
+        getNextPage: true,
+      })
+    )
+  }
+
+  const renderError = () => {
+    return (
+      <View testID={'PointsHistoryBottomSheet/ErrorState'} style={styles.errorContainer}>
+        <View style={styles.messageContainer}>
+          <Attention size={48} color={Colors.black} />
+          <Text style={styles.messageTitle}>{t('points.history.error.title')}</Text>
+          <Text style={styles.messageSubtitle}>{t('points.history.error.subtitle')}</Text>
+        </View>
+        <Button
+          testID={'PointsHistoryBottomSheet/TryAgain'}
+          onPress={onPressTryAgain}
+          text={t('points.history.error.tryAgain')}
+          type={BtnTypes.GRAY_WITH_BORDER}
+          size={BtnSizes.FULL}
+          style={{ width: '100%' }}
+        />
+      </View>
+    )
+  }
+
+  const sections = useMemo(() => {
+    if (!pointsHistory.length) {
+      return []
+    }
+    return groupFeedItemsInSections([], pointsHistory, (t: ClaimHistory) => Date.parse(t.createdAt))
+  }, [pointsHistory])
+
+  // TODO: Render items
+  const renderItem = () => {
+    return <></>
+  }
+
+  const renderHistory = () => {
+    return (
+      <View testID={'PointsHistoryBottomSheet/MainContent'} style={styles.contentContainer}>
+        <Text style={styles.contentHeader}>{t('points.history.title')}</Text>
+        <BottomSheetSectionList
+          renderItem={renderItem}
+          renderSectionHeader={(item) => (
+            <SectionHead text={item.section.title} style={styles.sectionHead} />
+          )}
+          sections={sections}
+          keyExtractor={(item) => `${item.activity}-${item.createdAt}`}
+          keyboardShouldPersistTaps="always"
+          testID="PointsHistoryList"
+          onEndReached={fetchMoreHistory}
+        />
+        {pointsHistoryStatus === 'loading' && (
+          <View style={styles.centerContainer}>
+            <ActivityIndicator
+              testID={'PointsHistoryBottomSheet/Loading'}
+              style={styles.loadingIcon}
+              size="large"
+              color={colors.primary}
+            />
+          </View>
+        )}
+      </View>
+    )
+  }
+
+  const renderContents = () => {
+    switch (pointsHistoryStatus) {
+      case 'idle':
+      case 'loading': {
+        return renderHistory()
+      }
+      case 'error': {
+        return renderError()
+      }
+      default: {
+        Logger.error(TAG, `Unknown points history status found: ${pointsHistoryStatus}`)
+      }
+    }
+  }
+
+  return (
+    <BottomSheetBase snapPoints={['80%']} forwardedRef={forwardedRef}>
+      <View style={styles.container}>{renderContents()}</View>
```

Comment:
i wonder if we could simplify this component by always having the BottomSheetFlatList here? i'm not sure if you've already tried and i'm overlooking some complexities....but naively i was thinking we could have the BottomSheetFlatList always, the sections would be always computed, the loader is always there as a child of the BottomSheetFlatList if the status is loading (i _think_ this makes it always appear at the bottom of the list). there is a ListEmptyComponent prop on the flatlist which you could use to create a component that displays either the error state or an empty state saying "you have no history" etc. depending on the history status. 

the advantage i see with this approach is that you can see at a glance all the components that could be present in the render, so you can easily understand what the screen could look like. it's personal preference but i find it much more difficult to visualise the screen by reading a render method that has many layers of logic handlers.

Reply:
OK, after messing around with this for a while, I think I've come to a working solution. Setting direct children of `BottomSheetSectionList` (or `SectionList` in general, I presume) doesn't work; they never rendered no matter what I did, and I"m guessing they're just ignored. Setting the `ListFooterComponent` however, did just the trick.

There's a _bit_ of complexity in selectively rendering the "My points activity" title (outside of the section list) depending on the history status, but that's not too bad - overall I think this approach is a whole lot cleaner!